pkg/store/query.go

```diff
@@ -0,0 +1,202 @@
+package store
+
+import (
+	"context"
+	"sync"
+
+	"io"
+
+	"github.com/go-kit/kit/log"
+	"github.com/improbable-eng/thanos/pkg/query"
```

Comment:
Maybe just make `storepb` a top-level package, move the query store stuff into `pkg/query` and the other implementations elsewhere respectively. Alternatively something like `pkg/store/{query,prometheus,bucket}`

Reply:
For the latter, we could move `query.StoreInfo` to `store.Info` maybe? That allows us to decouple `pkg/cluster` from the `pkg/store` as well and the `QueryStore` could become a generic `ProxyStore` – since otherwise there's no relationship to `pkg/query` which mainly refers to PromQL querying, right?

Reply:
Not sure if moving to many super small packages is really what we want, but maybe in this case when we have different implementation of single gRPC service.. That maybe make sense (:

`store.Info` will be used by `pkg/query` as well? or will you distinguish that? (totally separate struct)?  Not sure why it help with `pkg/cluster`. There is no such connection between `query.StoreInfo` and `pkg/cluster`

We can name QueryStore as ProxyStore even now :+1:  Also, all of these could be called just `Proxy` since it is in `store` package.

Reply:
`pkg/cluster` currently provides the StoreSet, which depends on `pkg/query`. But I'm actually moving this into `main` right now since it's really just a bit of glue. And adding static stuff to it makes it kind of awkward otherwise.